doc/pages/expansions.asciidoc

```diff
@@ -6,11 +6,33 @@ associated value before executing the command. These patterns are called
 expansions.
 
 Every expansion consists of a `%`, followed by the expansion _type_ (one
-or more alphabetic characters), a nestable punctuation character (`(`, `[`,
-`{`, or `<`), and then all the text up to and including its matching opposite
-(`)`, `]`, `}`, or `>`). It doesn't matter which character is used, but
-`{}` are most common. For example, the command `echo %val{session}` will
-echo the ID of the current Kakoune session.
+or more alphabetic characters), a quoting character, and then all the text
+up to and including its matching character.
+
+If an openning brace (`(`, `[`, `{`, or `<`) is used as the openning quoting
+character, the expansion will end at the character's matching brace (`)`, `]`,
+`}`, or `>`). Nested pairs of the braces used in the expansion are allowed, but
+they must be balanced. Braces other than the ones used in the expansion need
+not be balanced, however. For example, `%{nest{ed} non[nested}` is valid and
+expands to `nest{ed} non[nested`.
+
+If any other character is used, the expansion will end at the next occurance of
+that character. The quoting character can be escpaed inside the expansion if it
+is doubled-up. For example, `%|abc||def|` expands to the text `abc|def`.
+
+It doesn't matter which character is used, but `{}` are most common.
+
+There are 2 types of quoting which can be used to group together words separated
```

Comment:
I didn't realize the `command-parsing` page contained that (I don't think I've ever actually read it to be honest). The information on that page is what I though was missing from `expansions` (and ended up learning from looking at the source code). I just saw the minimal description and figured it was out-of-date or something. Should I remove the whole description of the syntax for expansions from the `expansions` page and leave it just for listing the types of expansions and linking to the `command-parsing` page?

Reply:
The `command-parsing` page is *mostly* about quoting, but touches on expansions because `%{}` is a kind of quoting and it would be weird to mention them at all.

This page is *mostly* about expansions, but touches on quoting because quoting can control when expansions happen, and because in shell-scripting you have to quote expansions all the time to keep them under control.

Probably all this information should be written down in one place and everything else just links to it, but Kakoune's documentation is more grown than built and it's not obvious where to put any given piece of information, or where to look for it (as you've discovered).

In the short term, probably the best thing is just to add a cross-reference to the `command-parsing` page.

Reply:
I added a cross-reference to `command-parsing`  and fixed a broken link to the `expansions` page.

After adding to `command-parsing` that %-strings are expanded inside double quotes (which applies to both typed expansions and regular strings), all of the quoting/escaping syntax and rules are in `command-parsing`, so the `expansions` page can just start with a link, the, and the thing about expanding to multiple arguments without leaving out any information.